### PR TITLE
#553 - Hide ImageName Field

### PR DESCRIPTION
**Pull request by waldroot**
Do not show image name field in license confirmation popup if image preview was saved before.

Now the user sees a confirmation popup without the ability to specify image name. 
I also stored the 'this.preview().buyCreditsUrl' in a variable in the right context as it changes in the 'confirm' function what returns 'TypeError: this.preview is not a function' when you click on 'Buy Credits' button.

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#<issue_number>: Issue title
2. ...

### Manual testing scenarios (*)

1. Login to admin panel
2. Open Magento Media Gallery (i.e. go to Cms -> Pages, edit the page and insert image)
3. Click "Search Adobe Stock" button to open images grid
4. Click on an image to expand the preview
5. Click on "Save Preview" button
6. Reopen images grid
7. Click on "License" button for the same image
You will see a confirmation popup without the ability to specify image name. If image hasn't been saved before then you will see the same popup but will be present and the input to specify the image name.

### CODE REVIEW

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js [hunk 1]
```diff
@@ -204,19 +204,20 @@ define([
...
                             {
-                                'title': $.mage.__('License Adobe Stock Image?'),
-                                'content': '<p>' + confirmationContent + '</p><p><b>' + quotaMessage + '</p><br>' +
-                                           $.mage.__('File Name') + '</b>',
-                                'visible': canPurchase,
+                                'title': $.mage.__('License Adobe Stock Images?'),
+                                'content': '<p>' + confirmationContent + '</p><p><b>' + quotaMessage + '</b></p><br>' + displayFieldName,
+                                'visible': !this.isDownloaded(),
                                 'actions': {
                                     confirm: function (fileName) {
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js:216]: `fileName` is `undefined` if input is not shown. The default generated filename should be used in that case, see `this.generateImageName(this.preview().displayedRecord())`

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @waldroot , thankyou for the pull request! I've spotted one issue in the implementation, can you please take a look at my review comment.

File: AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js (hunk 1 above)
Comment (waldroot) [reply AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js:216]: I have fixed the issue but i'm not really sure about my approach. Can you please check the code. Thank you.

**Comment by sivaschenko** [comment]
Hi @waldroot thanks for the pull request!

We are going to have a public meeting today at 20:00 CET. It will be great if you can join and demo your contribution!

Details: https://github.com/magento/adobe-stock-integration/wiki/Weekly-Public-Meeting

**Pull request merged**